src/model/element-base.ts

```diff
@@ -61,15 +61,16 @@ export abstract class ScannedElementBase implements Resolvable {
       }
       this.jsdoc =
           this.jsdoc ? jsdoc.join(commentJsdoc, this.jsdoc) : commentJsdoc;
+      this.applyJsdocDemoTags();
```

Comment:
I have been trying to find a way.  Originally these calls were in `resolve()` because they needed the document url to resolve the paths.   However, we all kind of talked about how resolve was supposed to be idempotent and not really supposed to mutate the scanned element and such so I was happy to pull these out of that method.

The problem is the demos need to be extracted when the jsdoc changes.  This happens in multiple places and there's no hook other than to make the call in the same places AFAIK.

I did pull it out of ScannedElementMixin where it doesn't do any good anyways because jsdoc isn't assigned.  So I have to keep the call in 3 places-- one in ScannedElementBase in the special case where we apply the html comment, and in ScannedPolymerElementMixin and ScannedPolymerElement after jsdoc is set.